src/discussions/topics/topic-group/topic/Topic.jsx

```diff
@@ -1,58 +1,72 @@
+/* eslint-disable no-unused-vars, react/forbid-prop-types */
 import { faQuestionCircle } from '@fortawesome/free-regular-svg-icons';
 import { faComments, faFlag } from '@fortawesome/free-solid-svg-icons';
-
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import PropTypes from 'prop-types';
 import React from 'react';
 import { useParams } from 'react-router';
 import { Link } from 'react-router-dom';
 import { Routes } from '../../../../data/constants';
 
-// eslint-disable-next-line no-unused-vars
-function Topic({ id, name, topics }) {
+function Topic({
+  id,
+  name,
+  subtopics,
+  questions,
+  discussions,
+  flags,
+}) {
   const { courseId } = useParams();
 
   return (
-    <div className="discussion-topic d-flex flex-column border-bottom pl-2 pt-1 pb-1" data-topic-id={id}>
-      <Link
-        className="topic-name"
-        to={
-          Routes.POSTS.PATH.replace(':discussionId', id)
-            .replace(':courseId', courseId)
-            .replace(':threadId', '')
-        }
-      >
+    <Link
+      className="discussion-topic d-flex flex-column list-group-item text-gray-900 text-decoration-none p-2"
+      data-topic-id={id}
+      to={
+        Routes.THREADS.PATH.replace(':courseId', courseId)
+          .replace(':topicId', id)
+      }
+    >
+      <div className="topic-name h6">
         { name }
-      </Link>
-      <div className="d-flex">
-        <span className="badge mr-1">
-          <FontAwesomeIcon icon={faQuestionCircle} />
-          22
-        </span>
-        <span className="badge mr-1">
-          <FontAwesomeIcon icon={faComments} />
-          33
+      </div>
+      <div className="d-flex lead">
+        <span className="badge mr-4">
+          <FontAwesomeIcon className="mr-2" icon={faQuestionCircle} />
+          {questions.length}
         </span>
-        <span className="badge">
-          <FontAwesomeIcon icon={faFlag} />
-          5
+        <span className="badge mr-4">
+          <FontAwesomeIcon className="mr-2" icon={faComments} />
+          {discussions.length}
         </span>
+        {flags.length > 0 && (
+          <span className="badge">
+            <FontAwesomeIcon className="mr-2" icon={faFlag} />
+            {flags.length}
+          </span>
+        ) }
       </div>
 
-    </div>
+    </Link>
   );
 }
 
 export const topicShape = {
   name: PropTypes.string,
   id: PropTypes.string,
-  topics: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
+  subtopics: PropTypes.array,
+  questions: PropTypes.arrayOf(PropTypes.object),
+  discussions: PropTypes.arrayOf(PropTypes.object),
+  flags: PropTypes.arrayOf(PropTypes.object),
```

Comment:
This approach is incorrect. This is now how the API will return the data. These will just be numbers.